src/bosh.js

```diff
@@ -632,6 +636,9 @@ Strophe.Bosh.prototype = {
                      this._requests[0].age() > Math.floor(Strophe.SECONDARY_TIMEOUT * this.wait))) {
                     this._restartRequest(0);
                 }
+
+                this._conn.ridChange(Number(req.rid) + 1);
+
```

Comment:
But you get in trouble if the script is aborted if you still have a pending request. It is mandatory to save the rid after a successful request, because the server does the same. Maybe you feel better if we rename the function to something like <code>nextValidRid</code>.